src/repositories/versionable/VersionableRepository.ts

```diff
@@ -22,9 +22,12 @@ export default class VersioningRepository<D extends mongoose.Document, M extends
         return await model.save();
     }
 
-    public count(query: any): Query<number> {
-        const finalQuery = { deleteAt: undefined, ...query };
-        return this.model.countDocuments(finalQuery);
+    // public count(query: any): Query<number> {
+    //     const finalQuery = { deleteAt: undefined, ...query };
+    //     return this.model.countDocuments(finalQuery);
+    // }
+    count = () => {
+        return this.model.countDocuments();
```

Comment:
this will return count of all the documents from the collection.
and we need to include those documents which are deleted.

Reply:
done